### DISCUSSION TITLE
Continue support for legacy admin theme?

### DISCUSSION THREAD

**Discussion by colinmollenhour** [OpenMage/magento-lts#4230] [Polls]
OpenMage has a new default theme thanks to the hard work of many, the development of which was supported by our community and served an important purpose to distinguish OpenMage from Magento as a new platform.

This theme was developed as an extension to the default theme and the default theme was left intact for backwards compatibility for those with lots of third-party extensions that needed to be updated. However, we now have an issue in that we have two themes to support and limited resources. In particular, there is ongoing disagreement that comes up when a new feature is added regarding if it is required that the old theme be updated as well. For example: https://github.com/OpenMage/magento-lts/issues/4217

My proposal would be to:
1. Officially declare the old theme as unsupported.
2. Create a PR for the next major version such that the old files are removed from the main repo.
3. (Optionally) Create a new repo for the community to maintain the old theme if they so choose. Effectively, maintaining compatibility will not be a requirement for new PRs on the main repo.

If you wish for the old theme to be supported, please indicate how or to what degree you are willing to help maintain it. To be clear, in (3) I am not volunteering to do this.

**Comment by sreichel** [comment]
There are some people that prefer the legacy theme (for some reasons i dont know), thats why i vote for number one.

From devs point of view ...

We dont have much changes to the admin template, so - for me - its not a valid argument to say it takes much more time to make PRs fit for both. There is a fallback logic - __that needs to be fixed__ (we have working PRs for it).

Lets take #4218 ... it took me a few minutes to copy&paste the template/css files and make minor adjustements.

>  in (3) I am not volunteering to do this.

Same. To work on different repos makes it harder

**Comment by addison74** [comment]
Do not forget that after installation the OpenMage theme is the default one, it is not necessary to change to legacy.

From personal experience, those who used Magento 1 continue to use the legacy theme in OpenMage. When I made the change, they asked for it back. Even I use this theme in my tests. Let's also analyze the fact that we could disappoint those who use the legacy theme and no longer have the possibility to put it back. 

I would not remove this Magento theme because of issues like #4217. We did not encounter any other problems so far, it simply works. If the vote result is for removing from the distribution, please open a repository so that it can be installed later by anyone who wants, as was done for some modules (e.g. Mage Backup).

As a personal opinion I don't like how the OpenMage theme looks, it could be better using a lot of CSS and JavaScript libraries like bootstrap, tailwind, animate and so on.

**Comment by empiricompany** [comment]
I agree, it's useless to waste time testing PR in both themes. Moreover, the legacy theme binds us to the old Magento 1 in terms of image without distinguishing us as better solution. 

As a first step it would be enough to remove the configuration to load the legacy theme, in the future it will be necessary to unify the styles of the Openmage package to the base theme that I am now rewriting or the styles and it's all a mess. 

Not supporting it officially allows us to not  waste time and resources in various PR, I also remember the integration of the new Tinymce and others PR which made me lose a lot of time. 

It is also true that our Openmage theme is ugly and messy, but this does not justify maintaining and having to maintain the old theme of Magento.

I had proposed a change a long time ago for a new theme, here https://github.com/OpenMage/magento-lts/discussions/3458, but it was forgotten by everyone.

**Comment by sreichel** [comment]
> I had proposed a change a long time ago for a new theme

I think there were some suggestions for new themes ... and none got real attention. :(

Maybe re-colletct them, add them to https://sreichel.github.io/magento-lts/users/modules/_themes/ (e.g. - and if it gets merged) and think about to make them part of core code.

I dont want to decide which should be part of core - and which not. There were a few good suggestions.

**Comment by colinmollenhour** [comment]
> As a personal opinion I don't like how the OpenMage theme looks, it could be better using a lot of CSS and JavaScript libraries like bootstrap, tailwind, animate and so on.

My concern is that with two themes to maintain, such improvements are too much work and so will never happen. If we get back to one theme, there is a chance to modernize it in major versions, but I don't imagine anyone wanting to do that to the new theme *and* the old M1 theme simultaneously.

Just like with PHP support, users will need to make some changes to their internal code to keep up. If there are only minor issues that are causing people to stick to the old theme vs adopt the new one, wouldn't you rather spend time fixing those than continuing to support the old theme?

**Reply by sreichel** [reply]
> As a personal opinion I don't like how the OpenMage theme looks, it could be better using a lot of CSS and JavaScript libraries like bootstrap, tailwind, animate and so on.

Dont want to judge about the look ...  (i am okay with it) ... just keep in mind years ago prototype was awesome. jquery was awesome. i worked with bootstrap that has (or had) a lot of overhead.

I am not that frontend/ui-guy .... i am open for anything, but would prefer plain js/css(scss) w/o any additional library.

**Comment by sreichel** [comment]
> Support should be dropped in the next major release with a separate repo for optional support. 

Question to the the up-voters ... 

Would you take the time to continue support? If not, better vote for keep it or drop it.

**Comment by ma4nn** [comment]
I voted for keeping the legacy admin theme, because (from a user's point of view) I don't not see any benefits of the current OpenMage admin theme:
- it is not responsive
- it does not look better or more up-to-date (yeah I know it's highly individual, but these blue or red buttons combined with the dreary background..🤨) 
- it uses way more space
- sure the legacy theme is not state-of-the-art, but to me it looks and feels more harmonious

From the comments here I reckon that more people are not a fan of the theme, so why then drop the alternative now instead of first providing a better one?

What about making a poll e.g. via Magento backend notifications and asking users what they want?

**Comment by hirale** [comment]
I think the new theme is pretty good, but the legacy is a bit outdated in style. If we want to drop the legacy theme, we can collect some opinions on the new theme before that to make it better.
